app/plugin_hooks_test.go

```diff
@@ -24,7 +24,8 @@ import (
 )
 
 func compileGo(t *testing.T, sourceCode, outputPath string) {
-	dir, err := ioutil.TempDir(".", "")
+	// Using ./pkg which is symlink to this package path
```

Comment:
Can you elaborate on why this is necessary?

Reply:
[This](https://github.com/mattermost/mattermost-server/blob/14d402005e45d3c83ac1dcd7f41527788c8f956e/app/plugin_hooks_test.go#L878) code requires a non-standard package called `github.com/pkg/errors`, which is present in the vendor folder. To be able to use that package, build needs to run from within mattermost-server path.

Another option would be instead of using a non-standard package, we can use standard `errors` package and not use any of the non-standard packages except packages defined in mattermost-server.

Reply:
3/5 on using the standard `errors` package there.

Reply:
Agree with @hanzei here, that we should probably simplify the code just to use `errors` and simplify the code. This will all go away once we adopt Go modules, anyway :)